external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSerdeSuite.scala

```diff
@@ -86,18 +86,18 @@ class AvroSerdeSuite extends SparkFunSuite {
     // deserialize should have no issues when 'bar' is nullable but fail when it is nonnull
     Deserializer.create(CATALYST_STRUCT, avro, BY_NAME)
     assertFailedConversionMessage(avro, Deserializer, BY_NAME,
-      "Cannot find non-nullable field 'foo.bar' (at position 0) in Avro schema.",
+      "Cannot find field 'foo.bar' (at position 0) in Avro schema at field 'foo'",
       nonnullCatalyst)
     assertFailedConversionMessage(avro, Deserializer, BY_POSITION,
-      "Cannot find non-nullable field at position 1 (field 'foo.baz') in Avro schema.",
+      "Cannot find field at position 1 ('foo.baz') in Avro schema at field 'foo'",
```

Comment:
at position 1 ('foo.baz') 
=>
'foo.bar' (at position 0)

Reply:
Sorry I don't understand, are you saying you think the expected message is incorrect? One thing to note is that this test is on `extraNonnullCatalyst` as the Catalyst schema, whereas the previous test is on `nonnullCatalyst`. We need an extra field because `CATALYST_STRUCT` and `nonnullCatalyst` have the same field count as `avro`, so when doing `BY_POSITION` it will succeed. Thus we use `extraNonnullCatalyst` which has an extra field `foo.baz` at position 1.

Reply:
ok I see. I didn't notice that the original error message has such a wording order too.